packages/core/src/modules/routing/RecipientApi.ts

```diff
@@ -370,8 +373,22 @@ export class RecipientApi {
     return mediationRecord
   }
 
-  public async notifyKeylistUpdate(connection: ConnectionRecord, verkey: string) {
-    const message = this.mediationRecipientService.createKeylistUpdateMessage(verkey)
+  public async notifyKeylistUpdate(connection: ConnectionRecord, verkey: string, action?: KeylistUpdateAction) {
```

Comment:
Not really related to this PR, but maybe we can update this method to something like:

```suggestion
  public async notifyKeylistUpdate({
    connectionId: string
    publicKeyBase58: string // or even publicKeyFingerprint
    action: KeylistUpdateAction // make action required
 }) {
 
 }
```

Reply:
Agree. I just didn't want to introduce breaking changes at the API level on this PR. But for sure this method must be updated. I would like to do something similar to Discover Features module, where we have the option of doing sync or async queries, in order to have a more harmonized Recipient API